In [1]:
import numpy as np

## Задание 1. Напишите уравнение параболы, проходящей через три точки (x,y): (1,2), (3,10), (5,1)

In [3]:
m = np.array([
    [1,1,1],
    [9,3,1],
    [25,5,1]
])
y = np.array([2,10,1])

In [4]:
np.linalg.solve(m,y)

array([-2.125, 12.5  , -8.375])

ответ - y = -2.125*x^2 + 12.5*x -8.375

## Задание 2. Известно, что свежий огурец на 99% состоит из воды. Месяц назад взвесили мешок со свежими огурцами. Получилось, что огурцов ровно 100 кг. Мешок убрали, а через месяц снова взвесили. Огурцы за это время усохли, и теперь вода составляет уже только 98% их веса. Сколько теперь (в кг) весят огурцы?


В первом звешивании масса самого огурца - 1% - 1 кг. Во втором 1 кг стал равен 2% - соотвественно все огурцы весят 50 кг.

## Задание 3. Определение логарифма. Решить уравнения:

1. 8
1. 8.229
1. log (8) ( 2 ^ (8x-4) = 4  
   1/3 * log (2) ( 2^ (8x-4) = 4 * log (2) (2)  
   2 ^ (8x-4) = 2 ^ 12  
   8x-4 = 12  
   x = 2  
1. 3 ^ (log (9) (5x-5)) = 5  
   log(3)(5) =1/2 * log(3)(5x-5)  
   2 * log(3)(5) = log(3)(5x-5)  
   log(3)(25) = log(3)(5x-5)  
   25 = 5x-5  
   x = 6  
1. x^(log(3)(x)+1) = 9  
   log(x)(9) = log(3)(x) + 1  
   2log(x)(3) = log(3)(x) + 1  
   2 / log(3)(x) = log(3)(x) + 1  
   log^2(3)(x) + log(3)(x) -2 = 0  
   y = log(3)(x)  
   y^2 + y - 2 = 0  
   y = 1, y = -2  
   log(3)(x) = 1, x = 3  
   log(3)(x) = -2, x = 1/9  
    

## Задание 4. Свойства логарифмов. Вычислить:

1. log(4)(16) = 2 * log(4)(4) = 2  
1. log(5)(1/25) = -2 * log(5)(5) = -2  
1. log(25)(5) = 1/2 * log(5)(5) = 1/2  
1. log(3)(27 ^ 0.5) = 0.5 * log(3)(3*9) = 0.5 * (1 + 2) = 1.5  
1. log(2)(12) - log(2)(3) = log(2)(12/3) = 2  
1. log(6)(12) + log(6)(3) = log(6)(36) = 2  
1. e ^ ln(5) = 5  
1. log(2)(225) / log(2)(15) = log(2)(15 ^ 2) / log(2)(15) = 2  
1. log(4)(32) + log(0.1)(10) = log(4)(32) + log(10 ^ -1)(10) =  
    log(4)(32) - 1 = log(4)(32) - log(4)(4) = log(4)(8) =  
    1/2 * log(2)(8) = 1/2 * 3 = 1.5  
1. 9 ^ (log(3)(5 ^ -2)) = 3 ^ (2 * 1/2 log(3)(5)) = 3 ^ (log(3)(5)) = 5